src/sitemaps.ts

```diff
@@ -0,0 +1,47 @@
+import * as dotenv from 'dotenv'
+import * as fsPromise from 'fs/promises';
+import { URL } from 'url';
+import { existsSync, readFileSync, unlinkSync } from 'fs';
+import { getStudiesAssess } from './helpers/studiesAssessment.js';
+import { dashLogger, logger } from "./helpers/logger.js";
+import { getStudiesFromSitemap } from "./helpers/fetchSitemapStudies.js";
+import { elasticIndexCheck } from './helpers/esFunctions.js';
+import { isFileEmpty } from './helpers/writeToFiles.js';
+dotenv.config({ path: '../.env' })
+
+//START SCRIPT EXECUTION
+logger.info('Start of Script');
+dashLogger.info(`Start of Script, time:${new Date().toUTCString()}`);
+//creates ES index if it doesnt exist, skips creating if it does exist
+await elasticIndexCheck();
+//check if failed.txt exists from previous run, and remove it for a clear run
+if (existsSync('../outputs/failed.txt'))
+  unlinkSync('../outputs/failed.txt');
+const file = await fsPromise.open('../inputs/sitemapsRUN.txt', 'r');
+//for each sitemap line of file input, begin loop
+for await (const sitemapLine of file.readLines()) {
+  logger.info(`Processing sitemap: ${sitemapLine}`);
+  dashLogger.info(`Processing sitemap: ${sitemapLine}, time:${new Date().toUTCString()}`);
+  const studiesAssessFiltered: string[] = await getStudiesFromSitemap(new URL(sitemapLine));
+  const outputName: string = new URL(sitemapLine).hostname;
+  await getStudiesAssess(studiesAssessFiltered, outputName);
+  logger.info(`Finished assessing sitemap: ${sitemapLine}`);
+  dashLogger.info(`Finished assessing sitemap: ${sitemapLine}, time:${new Date().toUTCString()}`);
+}
+//check file if any studies failed and re-assess them
+await isFileEmpty('../outputs/failed.txt')
```

Comment:
`../outputs/failed.txt` is removed at line 16, so this will never work as intended